chrome/content/zotero/xpcom/itemTreeManager.js

```diff
@@ -0,0 +1,370 @@
+	 *     iconPath: 'chrome://zotero/skin/tick.png',
+	 *     ignoreInColumnPicker: false, // show in the column picker
+	 *     submenu: true, // show in the column picker submenu
+	 *     primary: false, // only one primary column is allowed
+	 *     pluginID: 'make-it-red@zotero.org', // plugin ID, which will be used to unregister the column when the plugin is unloaded
+	 *     dataProvider: (item, dataKey) => {
+	 *         // item: the current item in the row
+	 *         // dataKey: the dataKey of the column
+	 *         // return: the data to display in the column
+	 *         return item.getField('title').split('').reverse().join('');
+	 *     },
+	 *     zoteroPersist: ['width', 'hidden', 'sortDirection'], // persist the column properties
+	 * });
+	 * ```
+	 * @example
+	 * Register multiple custom columns:
+	 * ```js
+	 * const registeredDataKeys = await Zotero.ItemTreeManager.registerColumns(
+	 * [
+	 *     {
+	 *          dataKey: 'rtitle',
+	 *          iconPath: 'chrome://zotero/skin/tick.png',
+	 *          label: 'reversed title',
+	 *          pluginID: 'make-it-red@zotero.org', // Replace with your plugin ID
+	 *          dataProvider: (item, dataKey) => {
+	 *              return item.getField('title').split('').reverse().join('');
+	 *          },
+	 *     },
+	 *     {
+	 *          dataKey: 'utitle',
+	 *          iconPath: 'chrome://zotero/skin/cross.png',
+	 *          label: 'uppercase title',
+	 *          pluginID: 'make-it-red@zotero.org', // Replace with your plugin ID
+	 *          dataProvider: (item, dataKey) => {
+	 *              return item.getField('title').toUpperCase();
+	 *          },
+	 *     },
+	 * ]);
+	 * ```
+	 */
+	async registerColumns(options) {
+		const registeredDataKeys = this._addColumns(options);
+		if (!registeredDataKeys) {
+			return false;
+		}
+		this._addPluginShutdownObserver();
+		await this._notifyItemTrees();
+		return registeredDataKeys;
+	}
+
+	/**
+	 * Unregister a custom column.
+	 * Although it's async, resolving does not promise the item trees are updated.
+	 * @param {string | string[]} dataKeys - The dataKey of the column to unregister
+	 * @returns {boolean} true if the column(s) are unregistered
+	 * @example
+	 * ```js
+	 * Zotero.ItemTreeManager.unregisterColumns('rtitle');
+	 * ```
+	 */
+	async unregisterColumns(dataKeys) {
+		const success = this._removeColumns(dataKeys);
+		if (!success) {
+			return false;
+		}
+		await this._notifyItemTrees();
+		return true;
+	}
+
+	// TODO: add cell renderer registration
+
+	/**
+	 * Get column(s) that matches the properties of option
+	 * @param {undefined | Partial.<ItemTreeColumnOptions> | Partial.<ItemTreeColumnOptions>[]} options - An option or array of options to match
+	 * @param {string[]} [types=["custom", "itemtree"]] - An array of column source type to include
+	 * @returns {ItemTreeColumnOptions[]}
+	 */
+	getColumns(options, types) {
+		types = types || ["custom", "itemtree"]
```

Comment:
This function should be renamed to `getCustomColumns()`, take one argument `itemTreeID` and return all custom registered columns that match `itemTreeID` or are `"*"`. Change the code that uses this function accordingly.

Reply:
Done. But no extra arg, because the `options` itself is used as a filter and just passes the `enabledTreeIDs` directly.